src/main/java/org/minions/devfund/marylin/Squarelotron.java

```diff
@@ -0,0 +1,275 @@
+            for (int j = 0; j < this.size; j++) {
+                this.squarelotron[i][j] = squarelotron[i][j];
+                this.squarelotron[i][j] = squarelotron[i][j];
+            }
+        }
+    }
+
+    /**
+     * Gets the Squarelotron grid.
+     *
+     * @return the Squarelotron grid.
+     */
+    public int[][] getSquarelotron() {
+        return squarelotron.clone();
+    }
+
+    /**
+     * Method to duplicate a grid.
+     *
+     * @param original original grid.
+     * @return returns the new grid.
+     */
+    private int[][] duplicateGrid(final int[][] original) {
+        int[][] backupGrid = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                backupGrid[i][j] = original[i][j];
+            }
+        }
+        return backupGrid;
+    }
+
+    /**
+     * Upside down flip the complete matrix.
+     *
+     * @param matrix grid to be upsideDownFlipCompleteMatrix.
+     * @return return the upsideDownFlipCompleteMatrix grid.
+     */
+    int[][] upsideDownFlipCompleteMatrix(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int temp = 0;
+        for (int i = size - 1; i >= size / 2; i--) {
+            for (int j = 0; j < size; j++) {
+                resultGrid[i][j] = matrix[temp][j];
+                resultGrid[temp][j] = matrix[i][j];
+            }
+            temp++;
+        }
+        return resultGrid;
+    }
+
+    /**
+     * Upside down flip the internal matrix.
+     *
+     * @param matrix grid to be upsideDownFlipInternalMatrix.
+     * @return return the upsideDownFlipInternalMatrix grid.
+     */
+    int[][] upsideDownFlipInternalMatrix(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int temp = 1;
+        for (int i = size - 2; i >= size / 2; i--) {
+            for (int j = 1; j < size - 1; j++) {
+                resultGrid[i][j] = matrix[temp][j];
+                resultGrid[temp][j] = matrix[i][j];
+            }
+            temp++;
+        }
+        return resultGrid;
+    }
+
+
+    /**
+     * Upside Down Flip given a ring.
+     *
+     * @param ring ring of the grid.
+     * @return the Squarelotron Upside Down Flip.
+     */
+    Squarelotron upsideDownFlip(int ring) {
+
+        //Returns the same squarelotron if the ring number is equal to the number of rings of a matrix n x n.
+        if (Math.abs(size) % 2 == 1 && ring == this.numberOfRings) {
+            return this;
+        }
+
+        int[][] resultGrid = upsideDownFlipHelper(ring);
+        return new Squarelotron(resultGrid);
+    }
+
+    /**
+     * upsideDownFlipHelper.
+     *
+     * @param ring ring number.
+     * @return a grid.
+     */
+    private int[][] upsideDownFlipHelper(int ring) {
+        int[][] resultGrid;
+        //If the ring is the internal rings.
+        if (ring != 1) {
+            resultGrid = upsideDownFlipInternalMatrix(this.squarelotron);
```

Comment:
If the matrix is 6x6 and the ring is 2.
It will flip all the values inside the ring, it should restore the inner values too.
For example this test will fail.
```java
@Test
public void testUpsideDownFlipixBySixGridSecondRing() {
    final int[][] expectedGrid = {{1,2,3,4,5,6}, {7,26,27,28,29,12},{13,20,15,16,23,18}, {19,14,21,22,17,24}, {25,8,9,10,11,30}, {31,32,33,34,35,36}};
    Squarelotron squarelotron = new Squarelotron(6);
    printGrid(squarelotron.getSquarelotron(), INITIALWORD);
    printGrid(expectedGrid, EXPECTEDWORD);
    Squarelotron actualSquarelotron = squarelotron.upsideDownFlip(TWO);
    printGrid(actualSquarelotron.getSquarelotron(), ACTUALWORD);
    assertTrue(ASSERTIONMESSAGE, Arrays.deepEquals(expectedGrid, actualSquarelotron.getSquarelotron()));
}
```